eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/LongInterval.java

```diff
@@ -0,0 +1,1092 @@
+/*
```

Comment:
[N.B. comment applies to whole class, so placed up here]
Why is the looping in this class so inconsistent? Sometimes it's fully expanded (e.g. `forEach`, `equals`, etc). Sometimes, it's using a for-loop with `this.get(i)`. Sometimes, it's using `forEach*` with a lambda. And yet other places, a `LongIterator` is used. It's like 4 different people wrote this class 😄

Reply:
Same as above - I followed the implementation pattern of `IntInterval` - one reason didn't want to diverge because it would make it harder to turn it into a code generated template if we decide to do so in the future.   My preference is to leave it as it and have a separate PR to clean up both classes (and keep them consistent). Please let me know what you think.

Reply:
My preference would be to have a clean implementation here and another PR to clean up IntInterval. Others are welcome to chime in. @donraab @nikhilnanivadekar @motlin ?

Reply:
I am more of let’s get this one in first and then clean up one or both. I am waiting for these PRs to be closer to release 10.3.0. So the fastest way to that is more preferable for me.

Reply:
I have fixed `sum()` and added more tests for it (specifically to address the concern about overflow during calculation), and have added a test for `isEmpty()`. My preference now would be to get `LongInterval` in as is and open a new issue for `LongInterval` and `IntInterval` code clean up (happy to work on it as the potential is certainly there!)